File: Mechanics of Materials - Computer Problems/chapter-1/p1_c4.m
%%%%%%%%% P1-C4 %%%%%%%%%%
clc; clear;

P = input("Force Applied [=> kips]: ");
ultimate_load = input("Ultimate Load [=> kips]: ");

c = 0;
alpha_range = cell(11, 1);
beta_range = cell(11, 1);

for i = atand(0:0.1:1)
    c = c + 1;
    val = i.* (pi/180);
    beta_range{c, 1} = val; alpha_range{c, 1} = val;
end

alpha_length = length(alpha_range);
beta_length = length(beta_range);

f_s = zeros(alpha_length, beta_length);

for i = 1:alpha_length
    for j = 1:beta_length
        alpha = alpha_range{i, 1};
        beta = beta_range{j, 1};

        force = P * ((15*sin(alpha) + 30 * cos(alpha)) / (15 * cos(beta) + 12 * sin(beta)));

        f_s(i, j) = ultimate_load / force;
    end
end

disp(f_s);

File: Mechanics_of_Materials-Computer-Problems/Code-Mechanics-of-Mat-Semester03/arshia/chapter-1/p1_c4.py
import math

P = float(input("force applied (kips)= "))
ultimate_load = float(input("ultimate load (kips)= "))

c = 0
alpha_range = [math.radians(i) for i in range(0, 11)]
beta_range = [math.radians(i) for i in range(0, 11)]

alpha_length = len(alpha_range)
beta_length = len(beta_range)

f_s = [[0] * beta_length for _ in range(alpha_length)]

for i in range(alpha_length):
    for j in range(beta_length):
        alpha = alpha_range[i]
        beta = beta_range[j]

        force = P * ((15 * math.sin(alpha) + 30 * math.cos(alpha)) / (15 * math.cos(beta) + 12 * math.sin(beta)))

        f_s[i][j] = round(ultimate_load / force, 3)

for row in f_s:
    print(row)